traitsui/qt4/instance_editor.py

```diff
@@ -122,10 +122,9 @@ def init(self, parent):
             self._choice.setReadOnly(True)
             self.set_tooltip(self._choice)
 
-        if droppable:
-            # Install EventFilter on control to handle DND events.
-            drop_event_filter = _DropEventFilter(self.control)
-            self.control.installEventFilter(drop_event_filter)
+            self.factory.observe(
+                self.rebuild_items, "values.items", dispatch="ui"
+            )
```

Comment:
The more I think about this it doesn't really make sense.  This was added because without it we were getting a `NotifierNotFound` error in the `non_editable_droppable_view` case.  However, the only time this occurs is the use case described here: https://github.com/enthought/traitsui/pull/1733/files#r672368160

In which case `values.items` should never change. (There isn't a list of instances, just the one!). 

Thinking about this more I think it probably makes more sense to check for this case when we arer removing the listener (and not try to remove it), rather than just adding it here to avoid the failure.

This code path is rather confusing.  Using `_choice` in this case _doesn't really make sense_.  At least not to me anyway, as we are not really choosing an instance from a list of options...